### PR TITLE
feature(raid_support): Support raid configuration

### PR DESCRIPTION
**Pull request by aleksbykov**
scylla-4.6 have support to create 2 kind of
raid configuration: RAID0 and RAID5.
Scylla: https://github.com/scylladb/scylla/commit/42fd73d0332d034809c6cd228785bff82e1845aa
Feature: https://github.com/scylladb/scylla/pull/9093

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by aleksbykov** [comment]
@dkropachev Could you review the commit. I didn't test it yet, because it depend on another https://github.com/aleksbykov/scylla-machine-image/commit/1a240fa4f98268fe91f54aa3233c57b0a5d47190

File: sdcm/sct_provision/aws/user_data.py [hunk 1]
```diff
@@ -39,6 +39,13 @@ def data_device(self) -> DataDeviceType:
             return DataDeviceType.ATTACHED.value
         return DataDeviceType.INSTANCE_STORE.value
 
+    @property
+    def raid_level(self) -> RaidLevelType:
+        """
+        Tell scylla setup to create RAID0 or RAID5 on disks
+        """
+        return self.params.get("raid_level", RaidLevelType.RAID0)
```
Comment (dkropachev) [inline sdcm/sct_provision/aws/user_data.py:47]: Wont work, you need to go wth:
```suggestion
        return self.params.get("raid_level") or RaidLevelType.RAID0
```

File: sdcm/provision/common/user_data.py [hunk 2]
```diff
@@ -59,3 +64,10 @@ def post_configuration_script(self) -> str:
     @abc.abstractmethod
     def to_string(self) -> str:
         pass
+
+    @property
+    @abc.abstractmethod
+    def raid_level(self) -> int:
```
Comment (dkropachev) [inline sdcm/provision/common/user_data.py:70]:
```suggestion
    def raid_level(self) -> RaidLevelType:
```

**Review by dkropachev — CHANGES_REQUESTED** [review]

**Comment by aleksbykov** [comment]
@dkropachev thanks, i didn't miss anything related to provision, did i?

**Comment by dkropachev** [comment]
> @dkropachev thanks, i didn't miss anything related to provision, did i?

Nope, you did a great job, should work like a charmed.

**Comment by dkropachev** [comment]
You can test it by looking at user-data, when machine-image is extracting it, just run `grep -i user-data -R ./db-logs`

**Comment by dkropachev** [comment]
Tested with proposed changes:

```
/var/log/syslog:Dec 16 08:34:23 ip-10-0-2-7 scylla_image_setup[992]: 2021-12-16 08:34:23,928 - [scylla_configure] - INFO - Got user-data: {"cluster_name": "longevity-10gb-3h-dkropachev-db-cluster-None", "data_device": "instance_store", "post_configuration_script": "IyEvYmluL2Jhc2gKCnNlZCAtaSAtZSAncy9eXCpbWzpibGFuazpdXSpzb2Z0W1s6Ymxhbms6XV0qbnByb2NbWzpibGFuazpdXSo0MDk2LyoJCXNvZnQJbnByb2MJCXVubGltaXRlZC8nICAgICAgICAgL2V0Yy9zZWN1cml0eS9saW1pdHMuZC8yMC1ucHJvYy5jb25mCmVjaG8gLWUgJyoJCWhhcmQJbnByb2MJCXVubGltaXRlZCcgPj4gL2V0Yy9zZWN1cml0eS9saW1pdHMuZC8yMC1ucHJvYy5jb25mCgpzZWQgLWkgJ3MvI01heFNlc3Npb25zIFwoLipcKSQvTWF4U2Vzc2lvbnMgMTAwMC8nIC9ldGMvc3NoL3NzaGRfY29uZmlnCnNlZCAtaSAncy8jTWF4U3RhcnR1cHMgXCguKlwpJC9NYXhTdGFydHVwcyA2MC8nIC9ldGMvc3NoL3NzaGRfY29uZmlnCnNlZCAtaSAncy8jTG9naW5HcmFjZVRpbWUgXCguKlwpJC9Mb2dpbkdyYWNlVGltZSAxNXMvJyAvZXRjL3NzaC9zc2hkX2NvbmZpZwpzeXN0ZW1jdGwgcmVzdGFydCBzc2hkCg==", "raid_level": 0, "scylla_yaml": {"cluster_name": "longevity-10gb-3h-dkropachev-db-cluster-None"}, "start_scylla_on_first_boot": false}

```

File: sdcm/sct_provision/aws/user_data.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/sct_provision/aws/user_data.py:47]: fixed

File: sdcm/provision/common/user_data.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/provision/common/user_data.py]: fixed

**Comment by dkropachev** [comment]
Alex, please rebase and lets merge it

**Comment by aleksbykov** [comment]
@dkropachev Conflicts were resolved

**Review by fgelcer — DISMISSED** [review]
LGTM

**Comment by roydahan** [comment]
Add a new longevity to use raid-5, try using it on GCE.
You can use an existing one + a configuration that just add raid-5.

**Comment by aleksbykov** [comment]
New longevity added.
Job will be added after commit merged

**Comment by aleksbykov** [comment]
Job: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/abykov/job/longevity-3h-10gb-gce-test/2/console. Instances created with raid 5

**Review by enaydanov — APPROVED** [review]

**Review by dkropachev — APPROVED** [review]

**Pull request merged**